ament_index_python/test/test_ament_index_python.py

```diff
@@ -181,4 +196,110 @@ def get_package_share_directory_test(package_name, expect_prefix):
     except PackageNotFoundError:
         pass
     except Exception as exc:
-        assert False, 'Expected PackageNotFoundError, got: {}'.format(type(exc))
+        assert False, 'Expected PackageNotFoundError, got: {}' \
+            .format(type(exc))
+
+
+def test_get_resource_types():
+    # Empty ament prefix will raise env error
+    set_ament_prefix_path([''])
+    try:
+        get_resource_types()
+    except EnvironmentError:
+        pass
+    except Exception as exc:
+        assert False, 'Expected EnvironmentError, got: {}'.format(type(exc))
+
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+    resources = get_resource_types()
+    assert resources == {'resource_type1',
+                         'resource_type2',
+                         'resource_type3',
+                         'resource_type4',
+                         'resource_type5',
+                         'packages'}, 'Expected resources to ' + \
+        'be found: resource_type1,  resource_type2, ' + \
+        'resource_type3,  resource_type4, ' + \
+        'resource_type5 and packages'
+
+    set_ament_prefix_path(['prefix1'])
+    resources = get_resource_types()
+    assert resources == {'resource_type1',
+                         'resource_type2',
+                         'resource_type4',
+                         'resource_type5',
+                         'packages'}, 'Expected resources to ' + \
+        'be found: resource_type1,  resource_type2, ' + \
+        'resource_type4, resource_type5 and packages'
+
+
+def test_main_tool(capsys):
+    set_ament_prefix_path(['prefix1', 'prefix2'])
+    base_path = os.path.dirname(__file__)
+
+    # Get output from the tool with name empty
+    main()
+    captured = capsys.readouterr()
+    expected_result = 'packages\n' + \
+                      'resource_type1\n' + \
+                      'resource_type2\n' + \
+                      'resource_type3\n' + \
+                      'resource_type4\n' + \
+                      'resource_type5\n'
+    assert captured.out == expected_result
+
+    # Get output from the tool with name empty
+    main(argv=['packages'])
+    captured = capsys.readouterr()
+    expected_result = 'bar\t' + base_path + os.sep + 'prefix1\n' + \
+                      'baz\t' + base_path + os.sep + 'prefix2\n' + \
+                      'foo\t' + base_path + os.sep + 'prefix1\n'
+    assert captured.out == expected_result
+
+    # Get output using both of the variables
+    main(argv=['packages', 'bar'])
+    captured = capsys.readouterr()
+    expected_result = base_path + os.sep + 'prefix1\n'
```

Comment:
Same about pathlib usage:
```suggestion
    expected_result = str(base_path / 'prefix1') + '\n'
```